### ISSUE TITLE
Inconsistent behavior for handling NaT in busday_offset

### ISSUE THREAD

**Issue by will133** [numpy/numpy#7021]
It seems like NaT is handled in a wrong way with busday_offset.  Here is an example running on numpy version 1.10.1:

```
In [1]: import numpy

In [2]: x = numpy.array([None, None], "M8[D]")

In [3]: x
array(['NaT', 'NaT'], dtype='datetime64[D]')

In [4]: numpy.busday_offset(x, 1, roll='nat')
array(['1970-01-01', '1970-01-01'], dtype='datetime64[D]')

In [5]: numpy.busday_offset(x, 1, roll='following')
array(['1970-01-01', '1970-01-01'], dtype='datetime64[D]')
```

It seems to treat the NaT as a valid int64 and roll with it forward.  I would expect that the returned values would always be NaT if roll is not 'raise'.  If you start of a NaT date, IMHO it should be treated like a nan and should produce nan as a result when you add anything to it.

**Comment by charris** [comment]
Hmm, that has changed in master, and not for the better
```
In [4]: x = numpy.array([None, None], "M8[D]")

In [5]: numpy.busday_offset(x, 1, roll='nat')
Out[5]: array(['257953538050-12-31', '257953538095-05-01'], dtype='datetime64[D]')
```
@mwiebe Could you take a look? @shoyer Might be of interest to you also.

**Comment by mwiebe** [comment]
I think the bug is here: 

https://github.com/numpy/numpy/blob/master/numpy/core/src/multiarray/datetime_busday.c#L291

```
    /* If we get a NaT, just return it */
    if (date == NPY_DATETIME_NAT) {
        return 0;
    }
```

It should be setting `*out = NPY_DATETIME_NAT;` before returning 0. Should be a pretty easy patch for anyone who wants to write a test and PR for it. :)

@charris the lack of setting `*out` likely explains the different values you're seeing.

**Comment by charris** [comment]
Thanks Mark.

**Comment by sam09** [comment]
Have a look at #7028

**Comment by charris** [comment]
Closed by #7028.
